packages/session-replay-browser/src/session-replay.ts

```diff
@@ -166,25 +189,28 @@ export class SessionReplay implements AmplitudeSessionReplay {
   }
 
   getShouldRecord(ignoreFocus = false) {
-    if (!this.identifiers || !this.config) {
+    if (!this.identifiers || !this.config || !this.identifiers.sessionId) {
       this.loggerProvider.error(`Session is not being recorded due to lack of config, please call sessionReplay.init.`);
       return false;
     }
+
+    if (!this.config.captureEnabled) {
+      this.loggerProvider.log(
+        `Session ${this.identifiers.sessionId} not being captured due to capture being disabled for project.`,
+      );
+      return false;
+    }
+
     const globalScope = getGlobalScope();
     if (!ignoreFocus && globalScope && globalScope.document && !globalScope.document.hasFocus()) {
-      if (this.identifiers.sessionId) {
-        this.loggerProvider.log(
-          `Session ${this.identifiers.sessionId} temporarily not recording due to lack of browser focus.`,
-        );
-      }
+      this.loggerProvider.log(
+        `Session ${this.identifiers.sessionId} temporarily not recording due to lack of browser focus.`,
+      );
       return false;
-    } else if (this.shouldOptOut()) {
-      if (this.identifiers.sessionId) {
-        this.loggerProvider.log(`Opting session ${this.identifiers.sessionId} out of recording due to optOut config.`);
-      }
-      return false;
-    } else if (!this.identifiers.sessionId) {
-      this.loggerProvider.warn(`Session is not being recorded due to lack of session id.`);
+    }
+
+    if (this.shouldOptOut()) {
```

Comment:
If capture is enabled but the user opts out, then we don't capture at all?

Reply:
Yep exactly!